src/biogeochem/CNBalanceCheckMod.F90

```diff
@@ -131,20 +135,26 @@ subroutine BeginCNGridcellBalance(this, bounds, cnveg_carbonflux_inst, &
     associate(                                                &
          begcb          => this%begcb_grc                   , &  ! Output: [real(r8) (:)]  (gC/m2) gridcell carbon mass, beginning of time step
          begnb          => this%begnb_grc                   , &  ! Output: [real(r8) (:)]  (gN/m2) gridcell nitrogen mass, beginning of time step
-         totc           => cnveg_carbonstate_inst%totc_grc  , &  ! Input:  [real(r8) (:)]  (gC/m2) total gridcell carbon, incl veg and cpool
-         totn           => cnveg_nitrogenstate_inst%totn_grc, &  ! Input:  [real(r8) (:)]  (gN/m2) total gridcell nitrogen, incl veg
+         totc           => soilbiogeochem_carbonstate_inst%totc_grc  , &  ! Input:  [real(r8) (:)]  (gC/m2) total gridcell carbon, incl veg and cpool
+         totn           => soilbiogeochem_nitrogenstate_inst%totn_grc, &  ! Input:  [real(r8) (:)]  (gN/m2) total gridcell nitrogen, incl veg
          c_cropprod1    => c_products_inst%cropprod1_grc    , &  ! Input:  [real(r8) (:)]  (gC/m2) carbon in crop products
          n_cropprod1    => n_products_inst%cropprod1_grc    , &  ! Input:  [real(r8) (:)]  (gC/m2) nitrogen in crop products
          c_tot_woodprod => c_products_inst%tot_woodprod_grc , &  ! Input:  [real(r8) (:)]  (gC/m2) total carbon in wood products
          n_tot_woodprod => n_products_inst%tot_woodprod_grc   &  ! Input:  [real(r8) (:)]  (gC/m2) total nitrogen in wood products
          )
 
     begg = bounds%begg; endg = bounds%endg
-
-    call cnveg_carbonflux_inst%hrv_xsmrpool_to_atm_dribbler%get_amount_left_to_dribble_beg( &
-         bounds, hrv_xsmrpool_amount_left_to_dribble(bounds%begg:bounds%endg))
-    call cnveg_carbonflux_inst%dwt_conv_cflux_dribbler%get_amount_left_to_dribble_beg( &
-         bounds, dwt_conv_cflux_amount_left_to_dribble(bounds%begg:bounds%endg))
+    
+    if(.not.use_fates_bgc)then
+       
+       call cnveg_carbonflux_inst%hrv_xsmrpool_to_atm_dribbler%get_amount_left_to_dribble_beg( &
```

Comment:
I'm not sure if we want to do dribbling in the future, I'm not exactly sure what it is, but I know we don't currently use it with FATES. If we do have that type of behavior with FATES in the future, it would be mediated through different data structures.  The key issue here is that the cnveg_carbonflux_inst array is not allocated.

Reply:
The dribbling has to do with the reconciling the C/N states after the yearly landuse change. So the change in state is "dribbled" out over the year rather than in one "shock" to the system. So I think you are right that FATES won't be handling it that way. I suppose for FATES this ties into the LUH2 work #1077. So resolving this issue.